apps/src/applab/designElements/elementUtils.js

```diff
@@ -88,15 +88,39 @@ module.exports.removeIdPrefix = function (element) {
   element.setAttribute('id', getId(element));
 };
 
+// TODO(dave): remove blacklist once element ids inside divApplab
+// are namespaced: https://www.pivotaltracker.com/story/show/113011395
+var ELEMENT_ID_BLACKLIST = [
+  'finishButton',
+  'submitButton',
+  'unsubmitButton',
+  'turtleImage',
+  'prompt-icon'
+];
+
+var TURTLE_CANVAS_ID = 'turtleCanvas';
+
 /**
- * Returns true if newId is available and won't collide with
- * other elements in design mode or the rest of applab.
+ * Returns true if newId is available and won't collide with other elements.
+ * Always reject element ids which are blacklisted or already exist outside divApplab.
+ * Allow or reject other element types based on the options specified.
  * @param {string} newId The id to evaluate.
- * @param {string} originalId The original id of this element.
- *     This will always be a valid value for newId.
+ * @param {string} originalId Optional original id of this element.
+ *     If present, this will always be a valid value for newId.
+ * @param {Object.<string, boolean>} options Optional map of options
+ *     indicating whether certain elements are allowed.
+ * @param {string} options.allowCodeElements allow element ids which are
+ *     inside divApplab. Default: false
+ * @param {string} options.allowDesignElements: allow element ids which, when
+ *     prefixed with "design_", already exist in designModeViz. Default: false
+ * @param {string} options.allowDesignPrefix: allow element ids which
+ *     start with "design_". Default: false
+ * @param {string} options.allowTurtleCanvas: allow turtle canvas element
+ *     to be created. Default: false
  * @returns {boolean}
  */
-module.exports.isIdAvailable = function(newId, originalId) {
+module.exports.isIdAvailable = function(newId, originalId, options) {
```

Comment:
This is actually probably fine for now, since a lot of this complexity will presumable go away when we do the longer term fix.